catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -423,6 +437,38 @@ protected SolrQuery getSolrQuery(QueryRequest request, SolrFilterDelegate solrFi
     return postAdapt(request, solrFilterDelegate, query);
   }
 
+  private SolrQuery getRealTimeQuery(SolrQuery originalQuery, Collection<String> ids) {
+    if (LOGGER.isDebugEnabled()) {
+      LOGGER.debug("originalQuery: {}", getQueryParams(originalQuery));
+    }
+    SolrQuery realTimeQuery = new SolrQuery();
+    for (Map.Entry<String, String[]> entry : originalQuery.getMap().entrySet()) {
+      if (entry.getKey().equals(CommonParams.Q)) {
```

Comment:
✏️ `CommonParams.Q.equals(entry.getKey())` would be safer though I doubt what you have would ever fail.

Reply:
done